clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -660,6 +670,8 @@ void LoopAttributes::clear() {
   PipelineDisabled = false;
   PipelineInitiationInterval = 0;
   SYCLNofusionEnable = false;
+  SYCLIntelFPGAPipelineEnable = false;
+  SYCLIntelFPGANPipelines = 0;
```

Comment:
Why do we need two variables to track this? Wouldn't `SYCLIntelFPGANPipelines > 0` imply that piplines are enabled?

Reply:
Thanks @AaronBallman for the reviews. We need two variables here:

/// Flag for llvm.loop.intel.pipelining.enable metadata.
   bool SYCLIntelFPGAPipelineEnable;

 /// Value for llvm.loop.intel.pipelining.enable metadata.
   unsigned SYCLIntelFPGANPipelines;

>>Wouldn't SYCLIntelFPGANPipelines > 0 imply that piplines are enabled?
Yes, this will enable pipeline. 

According to FE specification:
```
LVM IR
when applied on a loop, the LLVM IR should be attached to llvm.loop metadata:
!{!"llvm.loop.intel.pipelining.enable", i32 N}

The value of the !"llvm.loop.intel.pipelining.enable" metadata is an i32 value that corresponds to the boolean parameter N.
A value of 0 is used for not pipelining the loop.
A value of 1 is used for pipelining the loop.
```
 if we use only one variable like SYCLIntelFPGANPipelines > 0- this will only generate IR for N = 1 but no IR for N =0.
So, we need a second bool variable "SYCLIntelFPGAPipelineEnable" here so that we can get both 0 and 1 using variable "SYCLIntelFPGANPipelines"

Reply:
Ah, thank you for the explanation. I think using an `Optional` would be more clear. If the `Optional<bool>` has a value, then the `bool` corresponds to on or off, and if the optional has no value, then the attribute wasn't specified. WDYT?

Reply:
```if (const auto *IntelFpgaPipeline =
            dyn_cast<SYCLIntelFpgaPipelineAttr>(A)) {
      const auto *CE = cast<ConstantExpr>(IntelFpgaPipeline->getValue());
      Optional<llvm::APSInt> ArgVal = CE->getResultAsAPSInt();
      setSYCLIntelFPGAPipelineEnable();
      setSYCLIntelFPGANPipelines(ArgVal->getBoolValue() ? 1 : 0);
    }```
I  have this. Do you mean to use something like : Optional<bool> SYCLIntelFPGANPipelines?

Reply:
Nope, I meant something different. I'll comment below.

Reply:
>> Why do we need two variables to track this? 

I have added single var to trach this part.